### PR TITLE
[Preload] Fixes root path when rector installed as dependency

### LINKED ISSUE #6323: preload.php breaks bin/rector when installed as a dependency
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | v0.10.18              |
| Installed as   | composer dependency                           |

When I upgraded rector to the latest version, my build broke. It turned out that `bin/rector` requires `preload.php` which in turn requires some files from its `vendor` directory. Not **my** vendor directory (I installed rector as composer dependency) but **rector's** `vendor` directory e.g. 
```
require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Lexer.php';
```
which is then resolved as `{my_project_dir}/vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/Lexer.php` when rector is run.

image

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes #6323

### CODE REVIEW

**Comment by samsonasik** [comment]
@TomasVotruba this is a patch for preload.php used on rector installed as project dependency. I tested it locally and it working ok.

 The ECS error is unrelated, due to `php-cs-fixer` dependency

**Comment by TomasVotruba** [comment]
I see. The paths are correct, as it should load Rector's vendor, not project.

The preload.php should be included only on tests here or in prefixed version then

File: preload.php [hunk 1]
```diff
@@ -2,240 +2,263 @@
...
+            self::$rootPath = __DIR__;
+            return self::$rootPath;
+        }
+
+        self::$rootPath = __DIR__ . '/../../..';
+        return self::$rootPath;
+    }
+}
+
+RectorVendor::getRootPath() . '/vendor/nikic/php-parser/lib/PhpParser/Lexer.php';
```
Comment (alanbem) [inline preload.php:28]: Did you skip `require_once` on purpouse?

**Comment by alanbem** [comment]
> The preload.php should be included only on tests here or in prefixed version then

What if `composer` guys will come up with some standardized support for preload files? Is rector even kind of library that should use preload functionality?

**Comment by alanbem** [comment]
also should you preload files you do not own?

File: preload.php (hunk 1 above)
Comment (samsonasik) [reply preload.php:28]: Ooops, I missed that. As @TomasVotruba's comment https://github.com/rectorphp/rector/pull/6324#issuecomment-831855738 . It need only loaded on tests / prefixed version.

**Review by alanbem — APPROVED** [review]

**Comment by samsonasik** [comment]
@TomasVotruba implemented, I applied to tests bootstrap https://github.com/rectorphp/rector/pull/6324/commits/fc190180068ff0dd93a11b5d6eac86f6d3e6b6a7 and rector prefixed https://github.com/rectorphp/rector/pull/6324/commits/f4ec0a274eda9529af5c1509dd5c1dc6bd738ba4

**Pull request merged**